File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
       string a=strs[0];
       for(string b:strs){
        string answ="";
        
        for(int i=0;i<b.size()&&i<a.size();i++){
            if(b[i]==a[i]){
                 answ+=a[i];
                 continue;
                 
            }
            
            break;
            
            }
            a=answ;
        }
       
       return a;
    }
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        a=strs[0].strip()
        for b in strs:
            answ=[]
            mi=min(len(a),len(b))
            for k in range(mi):
                if a[k]==b[k]:
                    answ.append(a[k])
                else:    
                    break
            
            a=answ
        return "".join(a)

        